src/Event/FootprintListener.php

```diff
@@ -19,7 +19,7 @@ class FootprintListener implements EventListenerInterface
     protected $_defaultConfig = [
         'events' => [
             'Model.beforeFind' => -100,
-            'Model.beforeSave' => -100,
+            'Model.beforeRules' => -100,
```

Comment:
Having both `beforeRules` and `beforeSave` here would be redundant. Are you sure?
`beforeValidate` is plain wrong has nothing to do with this (see my comment d1cc82c34ffe4052e875b79b695fc319ca1a57b7).